bin/excite/excite.c

```diff
@@ -97,91 +97,20 @@ static char *rcs_id = "$Id: excite.c,v 1.25 2014/12/11 08:30:34 uratec Exp $";
 
 char *BOOL[] = { "FALSE", "TRUE" };
 
-/*  Command Name  */
-char *cmnd;
-
-
-void usage(int status)
-{
-   fprintf(stderr, "\n");
-   fprintf(stderr, " %s - generate excitation\n", cmnd);
-   fprintf(stderr, "\n");
-   fprintf(stderr, "  usage:\n");
-   fprintf(stderr, "       %s [ options ] [ infile ] > stdout\n", cmnd);
-   fprintf(stderr, "  options:\n");
-   fprintf(stderr, "       -p p  : frame period                  [%d]\n",
-           FPERIOD);
-   fprintf(stderr, "       -i i  : interpolation period          [%d]\n",
-           IPERIOD);
-   fprintf(stderr, "       -n    : gauss/M-sequence for unvoiced [%s]\n",
-           BOOL[GAUSS]);
-   fprintf(stderr, "                   default is M-sequence\n");
-   fprintf(stderr, "       -s s  : seed for nrand                [%d]\n", SEED);
-   fprintf(stderr, "       -h    : print this message\n");
-   fprintf(stderr, "  infile:\n");
-   fprintf(stderr, "       pitch period (%s)         [stdin]\n", FORMAT);
-   fprintf(stderr, "  stdout:\n");
-   fprintf(stderr, "       excitation (%s)\n", FORMAT);
-#ifdef PACKAGE_VERSION
-   fprintf(stderr, "\n");
-   fprintf(stderr, " SPTK: version %s\n", PACKAGE_VERSION);
-   fprintf(stderr, " CVS Info: %s", rcs_id);
-#endif
-   fprintf(stderr, "\n");
-   exit(status);
-}
-
-
-int main(int argc, char **argv)
+void excite(double* pitch, int n, double* out, int fprd, int iprd, Boolean gauss, int seed)
 {
-   int fprd = FPERIOD, iprd = IPERIOD, i, j, seed = SEED;
-   unsigned long next = SEED;
-   FILE *fp = stdin;
+   int i, j, k = 0;
+   unsigned long next = seed;
    double x, p1, p2, inc, pc;
-   Boolean gauss = GAUSS;
-
-   if ((cmnd = strrchr(argv[0], '/')) == NULL)
-      cmnd = argv[0];
-   else
-      cmnd++;
-   while (--argc)
-      if (**++argv == '-') {
-         switch (*(*argv + 1)) {
-         case 'p':
-            fprd = atoi(*++argv);
-            --argc;
-            break;
-         case 'i':
-            iprd = atoi(*++argv);
-            --argc;
-            break;
-         case 'n':
-            gauss = TR;
-            break;
-         case 's':
-            seed = atoi(*++argv);
-            --argc;
-            break;
-         case 'h':
-            usage(0);
-         default:
-            fprintf(stderr, "%s : Invalid option '%c'!\n", cmnd, *(*argv + 1));
-            usage(1);
-         }
-      } else
-         fp = getfp(*argv, "rb");
 
    if (gauss & (seed != 1))
       next = srnd((unsigned int) seed);
 
-   if (freadf(&p1, sizeof(p1), 1, fp) != 1)
-      return (1);
-
+   p1 = pitch[0];
    pc = p1;
 
-   for (;;) {
-      if (freadf(&p2, sizeof(p2), 1, fp) != 1)
-         return (0);
+   for (int fn = 1; fn < n; fn++) {
```

Comment:
I guess this line may cause compilation error on MSVC on windows. While I was working on https://github.com/r9y9/SPTK/pull/8, I found MSVC on AppVeyor throws error for C89-imcompatible codes.